### PR TITLE
2022 kie updates

### PR DESCRIPTION
**Pull request by originalname51**
To reduce load on the database and network we no longer need to save the compiled rule blob binary or drl string to the database. This is calculated for the rule engine on start up and on each time the recompile flag is set.

### CODE REVIEW

**Review by simbamarufu1 — COMMENTED** [review]
LGTM

File: gtas-parent/gtas-commons/src/main/java/gov/gtas/model/Message.java [hunk 1]
```diff
@@ -59,6 +59,9 @@ public Message() {
 	@ManyToMany
 	@JoinTable(name = "message_document", joinColumns = @JoinColumn(name = "document_id"), inverseJoinColumns = @JoinColumn(name = "message_id"))
 	private Set<Document> documents = new HashSet<>();
+
```
Comment (tjwdev) [inline gtas-parent/gtas-commons/src/main/java/gov/gtas/model/Message.java:62]: This resolves an issue where multiple messages exist on a passenger but there's no distinguishing way to determine which is the most recent. This prevents us from having to make a new model AND the impossible task of adding this tracking to the passenger model, which is immutable by design.

File: gtas-parent/gtas-commons/src/main/java/gov/gtas/model/PendingHitDetails.java [hunk 2]
```diff
@@ -221,4 +223,24 @@ public Boolean getLookout() {
     public void setLookout(Boolean lookout) {
         this.lookout = lookout;
     }
+
+	public static Set<PendingHitDetails> convertHits(Set<HitDetail> hits) {
+		Set<PendingHitDetails> phdSet = new HashSet<>();
+		for (HitDetail hd : hits) {
+			PendingHitDetails phd = new PendingHitDetails(hd.getHitEnum());
```
Comment (originalname51) [inline gtas-parent/gtas-commons/src/main/java/gov/gtas/model/PendingHitDetails.java:230]: More flexibility would be added if pending hit details converted both sets and single

File: gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/DroolFactGatheringThread.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
+package gov.gtas.job.scheduler;
```
Comment (originalname51) [inline gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/DroolFactGatheringThread.java:1]: Unused.

File: gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/HitPersistenceThread.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
+package gov.gtas.job.scheduler;
```
Comment (originalname51) [inline gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/HitPersistenceThread.java:1]: Unused

File: gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/HitDetailCreatorThread.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
+package gov.gtas.job.scheduler;
```
Comment (originalname51) [inline gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/HitDetailCreatorThread.java:1]: Unused

File: gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/RuleExecutionThread.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
+package gov.gtas.job.scheduler;
```
Comment (originalname51) [inline gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/RuleExecutionThread.java:1]: Unused.

File: gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/RuleRunnerScheduler.java [hunk 7]
```diff
@@ -97,45 +142,33 @@ public RuleRunnerScheduler(ApplicationContext ctx, MessageStatusRepository messa
...
-					logger.info("Done updating rule runner kie for " + kb.getKbName());
-				}
-			} else {
-				logger.info("making new rule kie for " + kb.getKbName());
-				addOrUpdateNameAndKie(kb);
-				logger.info("Done creating rule kie for " + kb.getKbName());
-			}
-		}
-
+	@Scheduled(fixedDelayString = "${ruleRunner.fixedDelay.in.milliseconds}", initialDelayString = "${ruleRunner.initialDelay.in.milliseconds}")
```
Comment (originalname51) [inline gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/RuleRunnerScheduler.java:166]: Differentiate scheduled times

File: gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/RuleRunnerScheduler.java [hunk 8]
```diff
@@ -233,17 +264,97 @@ public void ruleEngine() throws InterruptedException, IOException {
...
 	}
+
+	@JmsListener(destination = "GTAS_RULE_ENGINE", concurrency = "10")
+	public void ruleMediator(org.springframework.messaging.Message<?> message, javax.jms.Session session,
+			javax.jms.Message msg) throws JsonParseException, JsonMappingException, IOException {
+		ObjectMapper mapper = new ObjectMapper();
+		RuleResultsWithMessageStatus rrwms = mapper.readValue((String) message.getPayload(),
+				RuleResultsWithMessageStatus.class);
+		String previousQueue = rrwms.getQueueName();
+		String nextQueue = ruleMap.get(previousQueue);
```
Comment (tjwdev) [inline gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/RuleRunnerScheduler.java:285]: It is possible to enhance this combining fuzzyhits and normal hits.

File: gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/RuleRunnerThread.java [hunk 9]
```diff
@@ -36,6 +36,7 @@
 import gov.gtas.repository.AppConfigurationRepository;
```
Comment (originalname51) [inline gtas-parent/gtas-job-scheduler-war/src/main/java/gov/gtas/job/scheduler/RuleRunnerThread.java:36]: Deprecated

**Review by tjwdev — APPROVED** [review]
LGTM

File: gtas-parent/scripts/db/2.0/passenger_trip_detail_rollback.sql [hunk 10]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE passenger_trip_details
+drop  most_recent_message bigint(20) unsigned DEFAULT NULL;
+
+alter table passenger_trip_details
```
Comment (gtasautumn) [inline gtas-parent/scripts/db/2.0/passenger_trip_detail_rollback.sql:4]: this should be

```
alter table passenger_trip_details
    drop constraint FKgo03ewsl8acsta6odlb0ymuvl
```

**Pull request merged**